### PR TITLE
Add support for running flutter engine without UI

### PR DESCRIPTION
**Pull request by rwalczyna**
Adding support for non UI application for Tizen. I have added `FlutterCreateService` function, which initializes Engine without window.

I also work on change in flutter-tizen to add selector for creating service app (without UI).

### CODE REVIEW

**Comment by swift-kim** [comment]
Have you checked other platforms (such as windows and glfw) on how they implemented their headless mode? It will help you with API design and implementation. Note that you don't have to care about the API compatibilities, that is, that you don't necessarily have to keep the current Tizen embedder APIs as they are but can also change them if necessary.

**Review by swift-kim — APPROVED** [review]

**Comment by rwalczyna** [comment]
I've checked glfw implementation - as it is most similar to Tizen implementation. They are using open_gl config, but without gl_proc resolver. In our implementation it would require change to all open_gl methods (as there are no checks for empty renderer), so I decided to use software renderer - as it is not used anyway. I am only not sure whether name `FlutterCreateService` is the most suitable - I followed `FlutterCreateWindow` name.

glfw uses `FlutterDesktopRunEngine` so I don't know if `FlutterRunEngine` wouldn't be better instead of `FlutterCreateService`.

**Comment by swift-kim** [comment]
I think `FlutterRunEngine` or `FlutterEngineRun` (flutter_windows.h) would be better because

- It's more consistent with other platforms (otherwise you should add some API description for your new API)
- Flutter desktop APIs are usually platform app model agnostic, but "Service" is not

p.s. I'm going to deprecate `FlutterWindowProperties` soon and let the embedder decide the window size (there's some reason for this), so your implementation based on constructor overloading will need some change.

**Comment by rwalczyna** [comment]
I have changed function name to `FlutterEngineRun`. I think that the PR can wait till you update the embedder. Then I will adapt this commit to your change.

File: shell/platform/tizen/tizen_embedder_engine.cc [hunk 1]
```diff
@@ -138,40 +139,46 @@ bool TizenEmbedderEngine::RunEngine(
...
+    config.open_gl.present = Present;
+    config.open_gl.fbo_callback = GetActiveFbo;
+    config.open_gl.surface_transformation = Transformation;
+    config.open_gl.gl_proc_resolver = GlProcResolver;
+    config.open_gl.gl_external_texture_frame_callback =
+        OnAcquireExternalTexture;
+  } else {
+    config.type = kSoftware;
+    config.software.struct_size = sizeof(config.software);
+    config.software.surface_present_callback = SurfacePresentCallback;
```
Comment (swift-kim) [inline shell/platform/tizen/tizen_embedder_engine.cc:180]: How about this?
```suggestion
    config.software.surface_present_callback =
        [](void* user_data, const void* allocation, size_t row_bytes,
           size_t height) -> bool { return true; };
```

**Review by swift-kim — APPROVED** [review]
Generally looks good to me.

**Review by pwasowski2 — COMMENTED** [review]
LGTM 👌 (however, I'm not yet very familiar with embedder's code, so someone else should approve it)

File: shell/platform/tizen/tizen_embedder_engine.cc (hunk 1 above)
Comment (rwalczyna) [reply shell/platform/tizen/tizen_embedder_engine.cc]: I applied this suggestion, it simplifies code a bit.

**Review by pwasowski2 — COMMENTED** [review]
LGTM 👌

**Pull request merged**